src/viz.py

```diff
@@ -1,9 +1,273 @@
+                                   x=arrival_time,
+                                   y='cumEnergy_Wh',
+                                   bins=70,
+                                   cbar=True)
+    ax.set_title('Energy Delivered -- Arrival Time (per 10 mins)', fontsize=18)
+    ax.set_xlabel('Time in day', fontsize=14)
+    ax.set_ylabel('Energy Delivered in total (Wh)', fontsize=14)
+    plt.show()
+    # return f_enerDlv_arrvl
+
+
+def energyDelivered_Depttime_joint(df, departure_time):
+    """
+    This function is to draw a
+    df: Pandas DataFrame
+    departure_time: Pandas Series.
+    """
+    f_enerDlv_dept1 = sns.jointplot(data=df,
+                                    x=departure_time,
+                                    y='cumEnergy_Wh',
+                                    kind="hex",
+                                    color="#4CB391")
+    plt.show()
+    # return f_enerDlv_dept1
+
+
+def energyDelivered_Depttime(df, departure_time):
+    """
+    This function is to draw a
+    df: Pandas DataFrame
+    departure_time:
+    """
+    f_enerDlv_dept, ax = plt.subplots(figsize=(8, 5))
+    f_enerDlv_dept = sns.histplot(df,
+                                  x=departure_time,
+                                  y='cumEnergy_Wh',
+                                  palette="ch:s=-.2,r=.6",
+                                  bins=70,
+                                  cbar=True)
+    ax.set_title('Energy Delivered -- Departure Time (per 10 mins)', fontsize=18)
+    ax.set_xlabel('Time in day', fontsize=14)
+    ax.set_ylabel('Energy Delivered in total (Wh)', fontsize=14)
+    plt.show()
+    # return f_enerDlv_dept
+
+
+def energyDelivered_DurMaxChar(df):
+    """
+    This function is to draw a
+    df: DataFrame
+    """
+    # Delete one user's car with too large charging rate of 7,680 kW
+    # The charging power of Tesla Model Y according Tesla database:
+    # Charge Power = 11 kW AC    |    Fastcharge Power (max) = 250 kW DC
+    df = df.drop(df[(df['vehicle_maxChgRate_W'] == 7680000)].index)
+    f_enerDlv_durMaxChar, ax = plt.subplots(figsize=(10, 5))
+    sns.despine(f_enerDlv_durMaxChar, left=True, bottom=True)
+    f_enerDlv_durMaxChar = sns.scatterplot(data=df,
+                                           x="DurationHrs",
+                                           y="cumEnergy_Wh",
+                                           hue="vehicle_maxChgRate_W",
+                                           size="choice",
+                                           palette='crest',
+                                           sizes=(20, 6),
+                                           linewidth=0,
+                                           ax=ax)
+    ax.set_title('Energy Delivered -- Duration & Max Charging Rate', fontsize=18)
+    ax.set_xlabel('Duration Hours (h)', fontsize=14)
+    ax.set_ylabel('Energy Delivered (Wh)', fontsize=14)
+    plt.show()
+    # return f_enerDlv_durMaxChar
+
+
+# def energyDelivered_DurMaxChar(df):
```

Comment:
Is this function meant to be commented out?

Reply:
Yeah need to delete that function. I try to organize it later. Thanks.